### PR TITLE
Preliminary Haiku support

### PR DESCRIPTION
**Pull request by extrowerk**
zpqE

### CODE REVIEW

File: src/getExecutablePath.cpp [hunk 1]
```diff
@@ -134,6 +134,24 @@ std::string getExecutablePath()
     return ec ? "" : p.make_preferred().string();
 }
 
+#elif(BOOST_OS_HAIKU)
+/* #FIXME */
```
Comment (Flamefire) [inline src/getExecutablePath.cpp]: Probably can port this: https://github.com/rust-lang/rust/blob/83dfe7b27cf2debecebedd3b038f9a1c2e05e051/src/libstd/sys/unix/os.rs#L343

File: src/LocaleHelper.cpp [hunk 2]
```diff
@@ -64,6 +64,12 @@ std::locale createUtf8Locale()
     // Don't change the locale on OSX. Using "" fails with 'locale::facet::_S_create_c_locale name not valid'
     return LocaleHelper::getBfsDefaultLocale();
 }
+#elif BOOST_OS_HAIKU
+std::locale createUtf8Locale()
+{
+    // Don't change the locale on Haiku. Using "" fails with 'locale::facet::_S_create_c_locale name not valid'
```
Comment (Flamefire) [inline src/LocaleHelper.cpp]: Did you test this? Using `""` should work to get the system locale. Only apple is known to not give a sh*t

Comment (extrowerk) [reply src/LocaleHelper.cpp]: Yeah, this is required, same error on Haiku.

File: src/getExecutablePath.cpp (hunk 1 above)
Comment (extrowerk) [reply src/getExecutablePath.cpp]: Uh, this is cryptic to me.

Comment (Flamefire) [reply src/getExecutablePath.cpp]: It basically uses `_get_next_image_info` to fill a struct, checks for errors on that call and then accesses `info.name` which I assume is the exe path. I don't know Haiku or `_get_next_image_info` so you may know more. Maybe you can dig deeper. From a quick google search I assume there is a kernel header `image.h` or so which has these functions.

Comment (extrowerk) [reply src/getExecutablePath.cpp]: Will check, thank you!

**Pull request closed without merging**